integrations/pagerduty/main.py

```diff
@@ -30,17 +30,31 @@ def initialize_client() -> PagerDutyClient:
 async def enrich_service_with_analytics_data(
     client: PagerDutyClient, services: list[dict[str, Any]], months_period: int
 ) -> list[dict[str, Any]]:
-    async def fetch_service_analytics(service: dict[str, Any]) -> dict[str, Any]:
+    async def fetch_service_analytics(
+        services: list[dict[str, Any]]
+    ) -> list[dict[str, Any]]:
+        service_ids = [service["id"] for service in services]
         try:
-            analytics = await client.get_service_analytics(service["id"], months_period)
-            return {**service, "__analytics": analytics}
+            service_analytics = await client.get_service_analytics(
```

Comment:
```suggestion
            services_analytics = await client.get_service_analytics(
```

Reply:
How many services are we querying in a batch, should we limit it to some size? or does pg has limit for number of services to be queried?